.circleci/config.yml

```diff
@@ -23,7 +23,61 @@ jobs:
             brew install shellcheck
             bundle check || bundle install --jobs=4 --retry=3 --path .bundle
       - save_cache:
-          key: v1-gems-{{ checksum "Gemfile" }}
+          key: fastlane-macos-gems-{{ checksum "Gemfile" }}
+          paths:
+            - .bundle
+
+      - run:
+          name: Check PR Metadata
+          command: bundle exec danger || echo "danger failed"
+
+      - run: bundle exec fastlane test
+
+      - store_test_results:
+          path: /Users/distiller/test-results
+
+      - run:
+          name: Post Test Results to GitHub
+          command: bundle exec danger || echo "danger failed"
+          when: always # Run this even when tests fail
+  
+  "Ubuntu":
+    environment:
+      CIRCLE_TEST_RESULTS: $HOME/test-results
+      LC_ALL: C.UTF-8
+      LANG: C.UTF-8
+      FASTLANE_ITUNES_TRANSPORTER_PATH: .bundle
+    docker:
+      - image: circleci/ruby:2.3
+    shell: /bin/bash --login -eo pipefail
+    steps:
+      - checkout
+
+      - restore_cache:
+          key: fastlane-ubuntu-gems-{{ checksum "Gemfile" }}
+      - run:
+          name: Setup Build
+          command: |
+            mkdir -p $CIRCLE_TEST_RESULTS
+            echo "2.3" > .ruby-version
+            gem update --system
+            bundle check || bundle install --jobs=4 --retry=3 --path .bundle
+            sudo apt-get install shellcheck
+      - run:
+          name: Install additional dependency xar
+          command: |
+            cd /tmp
+            mkdir download
+            cd download
+            wget https://github.com/downloads/mackyle/xar/xar-1.6.1.tar.gz
```

Comment:
We might want to provide a pre-built binary to avoid having to do that every time and maybe host it here: https://launchpad.net/ubuntu/+ppas

You could of course host it on github first.

Reply:
That sounds like a good idea. I will look into that.

Reply:
Fascinating, there are actually 4 sources for ready built `xar`s there I didn't find googling before:
https://launchpad.net/ubuntu/+ppas?name_filter=xar
Unfortunately none of them are v1.6.1, so I will see if I can build this myself.

Reply:
https://launchpad.net/ubuntu/+ppas links to https://help.launchpad.net/Packaging/PPA links to https://help.launchpad.net/Packaging/PPA/Uploading links to https://help.launchpad.net/Packaging/PPA/BuildingASourcePackage links to http://packaging.ubuntu.com/html/ links to http://packaging.ubuntu.com/html/packaging-new-software.html - and is a bit complicated.

I think I will move this into a new issue and work on it independently from this PR. 

Do you think deploying it like this would be fine for the beginning @lacostej ?

Reply:
https://github.com/fastlane/fastlane/issues/11280 now exists.

Reply:
We should build a docker container that includes these (based on the circleci/ruby:2.3 container)

Reply:
See #11302